scripts/cert-manager/app-registration-cert-mananger.sh

```diff
@@ -0,0 +1,308 @@
+
+    rm $UPDATED_SECRET_VALUES_FILE
+fi
+
+#######################################################################################
+### Assign custom permission on TXT records
+###
+
+ROLENAME="DNS TXT Contributor"
+APP_ID=$(az role definition list --name "$ROLENAME" -o tsv)
+CRED_ROLE=true
+echo ""
+if [[ $APP_ID == "" ]]; then
+    echo "Role \"$ROLENAME\" does not exist."
+
+    if [[ $USER_PROMPT == true ]]; then
+        while true; do
+            read -p "Create Role $ROLENAME? (Y/n) " yn
+            case $yn in
+                [Yy]* ) break;;
+                [Nn]* ) CRED_ROLE=false; echo "Skipping."; break;;
+                * ) echo "Please answer yes or no.";;
+            esac
+        done
+    fi
+
+    printf "Creating app registration..."
+
+    if [[ $CRED_ROLE == true ]]; then
+        CUSTOMDNSROLE_JSON="cert-mananger-custom-dns-role.json"
+        test -f "$CUSTOMDNSROLE_JSON" && rm "$CUSTOMDNSROLE_JSON"
+    cat <<EOF >>${CUSTOMDNSROLE_JSON}
+    {
+        "Name": "$ROLENAME",
+        "Id": "",
+        "IsCustom": true,
+        "Description": "Can manage DNS TXT records only.",
+        "Actions": [
+            "Microsoft.Network/dnsZones/TXT/*",
+            "Microsoft.Network/dnsZones/read",
+            "Microsoft.Authorization/*/read",
+            "Microsoft.Insights/alertRules/*",
+            "Microsoft.ResourceHealth/availabilityStatuses/read",
+            "Microsoft.Resources/deployments/*",
+            "Microsoft.Resources/subscriptions/resourceGroups/read",
+            "Microsoft.Support/*"
+        ],
+        "NotActions": [
+        ],
+        "AssignableScopes": [
+            "/subscriptions/${AZ_SUBSCRIPTION_ID}/resourceGroups/${AZ_RESOURCE_GROUP_COMMON}/providers/Microsoft.Network/dnszones/${AZ_RESOURCE_DNS}"
+        ]
+    }
+EOF
+
+
+    az role definition create --role-definition ./cert-mananger-custom-dns-role.json
+    test -f "$CUSTOMDNSROLE_JSON" && rm "$CUSTOMDNSROLE_JSON"
+    fi
+else
+    echo -e "Role $ROLENAME exists.";
+fi
+#######################################################################################
+### Assign members to role
+###
+
+echo ""
+if [[ $USER_PROMPT == true ]]; then
+    while true; do
+        read -p "Assign '$APP_REGISTRATION_CERT_MANAGER' permission to $ROLENAME? (Y/n) " yn
+        case $yn in
+            [Yy]* ) DNSTXT_PERMISSIONS=true; break;;
+            [Nn]* ) echo "Skipping."; break;;
+            * ) echo "Please answer yes or no.";;
+        esac
+    done
+fi
+
+DEVELOPERS=$(az ad app list --filter "(displayName eq '$APP_REGISTRATION_CERT_MANAGER')" --query "[].appId" --output tsv)
+
+if [[ $DNSTXT_PERMISSIONS == true ]]; then
+    printf "Assigning role to app registration..."
+    UPDATED_DNS_PERMISSIONS=$(az role assignment create --assignee "$DEVELOPERS" --role "$ROLENAME" --scope "/subscriptions/${AZ_SUBSCRIPTION_ID}/resourceGroups/${AZ_RESOURCE_GROUP_COMMON}/providers/Microsoft.Network/dnszones/${AZ_RESOURCE_DNS}" 2>/dev/null)
```

Comment:
Delete the `$DEVELOPERS` variable and use the `$APP_ID` variable
```suggestion
if [[ $DNSTXT_PERMISSIONS == true ]]; then
    printf "Assigning role to app registration..."
    UPDATED_DNS_PERMISSIONS=$(az role assignment create --assignee "$APP_ID" --role "$ROLENAME" --scope "/subscriptions/${AZ_SUBSCRIPTION_ID}/resourceGroups/${AZ_RESOURCE_GROUP_COMMON}/providers/Microsoft.Network/dnszones/${AZ_RESOURCE_DNS}" 2>/dev/null)
```

Reply:
Deleted DEVELOPERS, and used --assignee "$APP_ID"